Tools/cases_generator/parser.py

```diff
@@ -214,8 +199,30 @@ def outputs(self) -> list[OutputEffect] | None:
 
     @contextual
     def output(self) -> OutputEffect | None:
+        return self.stack_effect()
+
+    @contextual
+    def cache_effect(self) -> CacheEffect | None:
+        # IDENTIFIER '/' NUMBER
+        if tkn := self.expect(lx.IDENTIFIER):
+            if self.expect(lx.DIVIDE):
+                num = self.require(lx.NUMBER).text
+                try:
+                    size = int(num)
+                except ValueError:
+                    raise self.make_syntax_error(f"Expected integer, got {num!r}")
+                else:
+                    return CacheEffect(tkn.text, size)
+
+    @contextual
+    def stack_effect(self) -> StackEffect | None:
+        # IDENTIFIER [':' IDENTIFIER]
+        # TODO: Arrays, conditions
         if tkn := self.expect(lx.IDENTIFIER):
-            return StackEffect(tkn.text)
+            type = ""
+            if self.expect(lx.COLON):
+                type = self.require(lx.IDENTIFIER).text
+            return StackEffect(tkn.text, type)
```

Comment:
Not a huge fan of shadowing `type` here:

```suggestion
            type_ = ""
            if self.expect(lx.COLON):
                type_ = self.require(lx.IDENTIFIER).text
            return StackEffect(tkn.text, type_)
```

Reply:
Not a fan of suffix `_`, but I hear you, and I can change this to `typ`. I'll make a pass.